.coderabbit.yaml

```diff
@@ -0,0 +1,353 @@
+  collapse_walkthrough: false
+  
+  # Include a summary of which files were changed and how in the walkthrough
+  changed_files_summary: true
+  
+  # Generate sequence diagrams to visualize code flow in the walkthrough
+  sequence_diagrams: true
+  
+  # Estimate and display the code review effort/complexity
+  estimate_code_review_effort: true
+  
+  # Analyze how well the PR addresses any linked GitHub issues
+  assess_linked_issues: true
+  
+  # Include potentially related issues in the walkthrough
+  related_issues: true
+  
+  # Include potentially related PRs in the walkthrough
+  related_prs: true
+  
+  # Suggest appropriate labels based on the changes
+  suggested_labels: true
+  
+  # Automatically apply suggested labels (disabled - let maintainers control)
+  auto_apply_labels: false
+  
+  # Suggest reviewers based on code ownership and expertise
+  suggested_reviewers: true
+  
+  # Automatically assign suggested reviewers (disabled - let maintainers control)
+  auto_assign_reviewers: false
+  
+  # Disable poem generation in walkthrough comments
+  poem: false
+  
+  # Focus reviews on source code, exclude generated and vendor files
+  path_filters:
+    # Include source code
+    - "**/*.go"
+    - "**/*.yaml"
+    - "**/*.yml"
+    - "**/*.md"
+    - "**/*.proto"
+    - "**/Dockerfile*"
+    - "**/Earthfile"
+    - "**/*.sh"
+    
+    # Exclude generated and vendor files
+    - "!**/zz_generated*.go"
+    - "!**/vendor/**"
+    - "!**/node_modules/**"
+    - "!**/*.pb.go"
+    - "!**/*.pb.gw.go"
+    - "!**/mock_*.go"
+    - "!**/fake/**"
+    - "!**/testdata/**"
+    - "!**/dist/**"
+    - "!**/build/**"
+    - "!**/.tmp-earthly-out/**"
+  
+  # Path-specific instructions for different areas of the codebase
+  path_instructions:
+    - path: "**/*.go"
+      instructions: |
+        Enforce Crossplane-specific patterns: Use crossplane-runtime/pkg/errors
+        for wrapping. Check variable naming (short for local scope, descriptive
+        for wider scope). Ensure 'return early' pattern. Verify error scoping
+        (declare in conditionals when possible). For nolint directives, require
+        specific linter names and explanations. CRITICAL: Ensure all error
+        messages are meaningful to end users, not just developers - avoid
+        technical jargon, include context about what the user was trying to do,
+        and suggest next steps when possible.
+    
+    - path: "**/*_test.go"
+      instructions: |
+        Enforce table-driven test structure: PascalCase test names (no
+        underscores), args/want pattern, use cmp.Diff with
+        cmpopts.EquateErrors() for error testing. Check for proper test case
+        naming and reason fields. Ensure no third-party test frameworks (no
+        Ginkgo, Gomega, Testify).
+    
+    - path: "**/*.md"
+      instructions: |
+        Ensure Markdown files are wrapped at 80 columns for consistency and
```

Comment:
can we make this a bit wider? i've always thought 80 was unnecessarily constraining. would 100 be okay?